src/js/ui/search/searchView.tsx

```diff
@@ -0,0 +1,238 @@
+            </div>
+          </div>
+        <button key="search" className="newGameButton" type="submit">
+          <span className="fa fa-search" />
+          {i18n('search')}
+        </button>
+      </form>
+      {ctrl.result() ?
+        <div className="searchGamesList">
+          {ctrl.games().map((g: UserGameWithDate, index: number) => m(Game, { key: g.id, g, index, boardBounds, ctrl })) }
+          {ctrl.result().paginator.nextPage ?
+            <button key="more" className="newGameButton" oncreate={helper.ontap(() => ctrl.more())}>
+              <span className="fa fa-arrow-down" />
+            </button>
+            : null}
+        </div>
+      : null }
+    </div>
+  );
+}
+
+function renderSelectRow(label: string, isDisplayed: boolean, select1: Select, select2: Select) {
+  if(!isDisplayed) return null;
+
+  return (
+    <div className="game_search_row">
+      <label>{label}: </label>
+      <div className={'game_search_select' + (select2 ? '' : ' double_wide')}>
+        <select id={select1.name.replace('.', '_')} name={select1.name} onchange={select1.onchange ? select1.onchange : ''}>
+          { select1.default !== null ?
+            <option selected value=""> {select1.default} </option> :
+            null
+          }
+          {select1.options.map(renderOption)}
+        </select>
+      </div>
+      {select2 ?
+        <div className="game_search_select">
+          <select id={select2.name.replace('.', '_')} name={select2.name} onchange={select1.onchange ? select1.onchange : ''}>
+            { select2.default !== null ?
+              <option selected value=""> {select2.default} </option> :
+              null
+            }
+            {select2.options.map(renderOption)}
+          </select>
+        </div>
+        : null }
+    </div>
+  )
+}
+
+function renderOption(opt: Option) {
+  return (
+    <option key={opt.value} value={opt.value}> {opt.label} </option>
+  )
+}
+
+function getPlayers(): Array<Option> {
+  const playerAEl = (document.getElementById('players_a') as HTMLInputElement);
+  const playerBEl = (document.getElementById('players_b') as HTMLInputElement);
+  if(!playerAEl || !playerBEl)
+    return [];
+  const playerA = playerAEl.value.trim();
+  const playerB = playerBEl.value.trim();
+  const players: Array<Option> = [];
+  if (playerA) {
+    players.push({value: playerA, label: playerA});
+  }
+  if (playerB) {
+    players.push({value: playerB, label: playerB});
+  }
+  return players;
+}
+
+function playersNonEmpty() {
+  return getPlayers().reduce((acc, cur) => acc + cur.label.length, 0) > 0;
+}
+
+function isComputerOpp() {
+  const hasAi = (document.getElementById('hasAi') as HTMLInputElement);
+  return hasAi && (hasAi.value === '1');
+}
+
+const Game: Mithril.Component<{ g: UserGameWithDate, index: number, boardBounds: {height: number, width: number}, ctrl: SearchState }, { boardTheme: string }> = {
```

Comment:
This is the same component as in userGames.tsx. It should be extracted, put in `shared` folder.